heare/developer/context.py

```diff
@@ -78,6 +104,10 @@ def with_user_interface(self, user_interface: UserInterface) -> "AgentContext":
             user_interface=user_interface,
             usage=self.usage,
             memory_manager=self.memory_manager,
+            chat_history=self.chat_history.copy() if self.chat_history else [],
```

Comment:
this method is only used for sub agents (correct me if I'm wrong). as such, it should not propagate chat_history and tool_result_buffer.